flink-table/flink-sql-parser/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -2427,6 +2444,8 @@ SqlDrop SqlDropExtended(Span s, boolean replace) :
     (
         drop = SqlDropCatalog(s, replace)
         |
+        drop = SqlDropMaterializedTable(s, replace)
```

Comment:
We should throw an exception if the user specifies the TEMPORARY keyword, similar to the `CREATE MATERIALIZED TABLE` process logic.